### PR TITLE
Kylepressel/activation

### PR DESCRIPTION
**Pull request by pressel**
This is a replacement PR for the aerosol activation. 

This PR adds a new feature to the Fortran implementation of P3 microphysics. In particular, it allows the number of aerosol particles that are activated to form cloud liquid droplets and ice particles, as predicted by the aerosol scheme, to be included as source terms to the cloud and ice number concentrations that are prognosed as part of P3 microphysics. This feature is turned on by default.

The implementation of the feature amounts to adding two source terms to the right hand sides of the prognostic equations for nitot and nc that depend on naii and npccn, respectively. Both naii and npccn are predicted by the aerosol scheme.

These modifications to the source code are a non-BFB change.

### CODE REVIEW

File: components/cam/src/physics/cam/micro_p3.F90 [hunk 1]
```diff
@@ -1276,54 +1284,73 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
           endif
```
Comment (AaronDonahue) [inline components/cam/src/physics/cam/micro_p3.F90]: Should this chunk of code be removed, considering the repeated loop below?

File: components/cam/src/physics/cam/micro_p3.F90 [hunk 2]
```diff
@@ -674,6 +676,12 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
              birim(i,k) = 0.
           endif
 
+
+         !Activaiton of cloud droplets 
+          if (log_predictNc) then 
+             nc(i,k) = nc(i,k) + npccn(i,k) * dt
+          endif 
+
```
Comment (AaronDonahue) [inline components/cam/src/physics/cam/micro_p3.F90]: This addition isn't in the original PR #72 , does this reflect one of the newer changes requested by Peter C?

File: components/cam/src/physics/cam/micro_p3.F90 [hunk 3]
```diff
@@ -1651,7 +1678,7 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
           qr(i,k) = qr(i,k) + (qcacc+qcaut+qrcon-qrevp)*dt
 
           if (log_predictNc) then
-             nc(i,k) = nc(i,k) + (-ncacc-ncautc+ncslf+ncnuc)*dt
+             nc(i,k) = nc(i,k) + (-ncacc-ncautc+ncslf)*dt
```
Comment (AaronDonahue) [inline components/cam/src/physics/cam/micro_p3.F90]: Similarly, this removal of ncnuc isn't in PR #72 .

File: components/cam/src/physics/cam/micro_p3.F90 (hunk 2 above)
Comment (pressel) [reply components/cam/src/physics/cam/micro_p3.F90]: That is correct, this was in response to Peter C comment  on PR #72 about where droplet nucleation should occur. I should have made that more clear.

File: components/cam/src/physics/cam/micro_p3.F90 (hunk 3 above)
Comment (pressel) [reply components/cam/src/physics/cam/micro_p3.F90]: This was also in response to Peter C's comment on PR #72 about where nucleation should occur.

**Pull request merged**